src/discord/staff/censor.py

```diff
@@ -30,6 +30,12 @@ def __init__(self, bot: PiBot):
         default_permissions=discord.Permissions(manage_messages=True),
     )
 
+    @classmethod
```

Comment:
Why make this into a `classmethod`? You already have a reference to `self` in the places where it is used, so just make it a normal method.

Reply:
The function is pure and doesn't rely on an instance of the Cog to function correctly, hence why it's a static method.

I'd rather keep it that way since it is clear that `self` (i.e. you don't need to have, or construct, an instance of the cog) is not needed to call the method.

Reply:
But it's being used only inside the censor class, since it's responsible for the censoring. I understand where you're coming from, but I don't think this method will be used outside of this class.

Reply:
I still fail to see why it has to be an instance method since it does not access the Cog's internal state. Like I said, the function is entirely pure. I would even say that it is an anti-pattern to mark a function as an instance method if it could have instead been marked as static (sorta going off of a Principle of Least Privilege standpoint).

In general, I would structure it like this regardless of whether the function was being used in either a private or public scope.

Reply:
I appreciate your reasoning, but in cases like this (where there are disagreements about how the code should be structured), I usually just defer to three style guides, and we can follow what they say:

- PEP 8: Nothing about `@classmethod`
- Hitchhiker's: Nothing about `@classmethod`
- Google: [Use `classmethod` only when writing a named constructor, or a class-specific routine that modifies necessary global state such as a process-wide cache.](https://google.github.io/styleguide/pyguide.html#2174-decision)

What do you think?

Reply:
Ok, so reading up on google styleguide, I'm reconsidering using `@classmethod` in this case, since the function is _supposed_ to be static. The styleguide does mention not to use `@staticmethod`, but rather a module-level function, so should we just revert this back to [e22bb62](https://github.com/cbrxyz/pi-bot/commit/e22bb625fdb90df0121eee94d3e1d79ad7b3e442)?

https://github.com/cbrxyz/pi-bot/compare/e22bb625fdb90df0121eee94d3e1d79ad7b3e442..6dabe54b9e9a8944f363b3263a93a1e9e287deb5

Reply:
Not sure why you'd want to put it in the global namespace for every other part of the bot's code to access when this function is only going to be used in the censor functionality... not going to fight over it or anything, but not the biggest fan..